pkg/storage/unified/sql/backend.go

```diff
@@ -215,28 +216,46 @@ func (b *backend) initPruner(ctx context.Context) error {
 		MaxWait:    time.Minute * 5,
 		ProcessHandler: func(ctx context.Context, key pruningKey) error {
 			return b.db.WithTx(ctx, ReadCommitted, func(ctx context.Context, tx db.Tx) error {
-				res, err := dbutil.Exec(ctx, tx, sqlResourceHistoryPrune, &sqlPruneHistoryRequest{
-					SQLTemplate:  sqltemplate.New(b.dialect),
-					HistoryLimit: defaultPrunerHistoryLimit,
+				req := &sqlPruneHistoryRequest{
+					SQLTemplate: sqltemplate.New(b.dialect),
 					Key: &resource.ResourceKey{
 						Namespace: key.namespace,
 						Group:     key.group,
 						Resource:  key.resource,
 						Name:      key.name,
 					},
-				})
+					Generation:   true,
+					HistoryLimit: 1, // Only one version for each generation
+				}
+
+				res, err := dbutil.Exec(ctx, tx, sqlResourceHistoryPrune, req)
```

Comment:
NOTE:  I think we should leave this part out of the PR before merging, and make a new PR that uses the generation for pruning.

This is here so we can see that it works 🎉